atoma-state/src/handlers.rs

```diff
@@ -1550,6 +1613,76 @@ pub(crate) async fn handle_node_metrics_registration_event(
     let gpu_temperatures: Vec<_> = gpus.iter().map(|gpu| gpu.temperature).collect();
     let gpu_power_usages: Vec<_> = gpus.iter().map(|gpu| gpu.power_usage).collect();
 
+    if num_gpus > 0
+        && (gpu_memory_used.len() != num_gpus as usize
+            || gpu_memory_total.len() != num_gpus as usize
+            || gpu_memory_free.len() != num_gpus as usize
+            || gpu_percentage_time_read_write.len() != num_gpus as usize
+            || gpu_percentage_time_execution.len() != num_gpus as usize
+            || gpu_temperatures.len() != num_gpus as usize
+            || gpu_power_usages.len() != num_gpus as usize)
+    {
+        error!(
+            target = "atoma-state-handlers",
+            event = "handle-node-metrics-registration-event",
+            "Invalid GPU metrics:\n\
+             Expected {num_gpus} GPUs, got:\n\
+             - {} GPU memory used\n\
+             - {} GPU memory total\n\
+             - {} GPU memory free\n\
+             - {} GPU percentage time read/write\n\
+             - {} GPU percentage time execution\n\
+             - {} GPU temperatures\n\
+             - {} GPU power usages",
+            gpu_memory_used.len(),
+            gpu_memory_total.len(),
+            gpu_memory_free.len(),
+            gpu_percentage_time_read_write.len(),
+            gpu_percentage_time_execution.len(),
+            gpu_temperatures.len(),
+            gpu_power_usages.len()
+        );
+        return Err(AtomaStateManagerError::InvalidGpuMetrics(format!(
```

Comment:
We probably want to continue the trend of having granular errors as we do in other instances

e.g.

```rust
#[derive(Debug)]
enum PerformanceError {
    NoGpusAvailable,
    InvalidMetrics(String),
    DatabaseError(String),
    CalculationError(String),
}
```

Reply:
Yeah, totally in favor of this

Reply:
https://github.com/atoma-network/atoma-proxy/issues/233